spec/API_specification/index.rst

```diff
@@ -11,6 +11,7 @@ API specification
    array_object
    indexing
    data_types
+   data_type_functions
```

Comment:
We should either move `result_type` into this new category, or not create it.

Reply:
I'm planning to add `can_cast `, ` broadcast_arrays ` and `broadcast_to` to this same file, should I move `result_type` to this file in this PR?

Reply:
Yes please!